pyteomics/mztab.py

```diff
@@ -744,8 +744,8 @@ def _parse(self):
                 self.small_molecule_evidence_table.add(tokens[1:])
 
     def _determine_schema_version(self):
-        version_parsed, variant = re.search(r"(?P<schema_version>\d+.\d+.\d+)(?:-(?P<schema_variant>[MP]))?", self.version).groups()

```

Comment:
We can keep the regex for defending against non-integer matches using non-capturing optional groups:
`(?P<schema_version>\d+(?:.\d+(?:.\d+)?)?)(?:-(?P<schema_variant>[MP]))?`